libraries/plugins/net_stats/net_stats.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2018 net_stats and contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <graphene/net_stats/net_stats.hpp>
+#include <graphene/net/node.hpp>
+
+#include <prometheus/exposer.h>
+#include <prometheus/registry.h>
+#include <prometheus/counter.h>
+#include <prometheus/histogram.h>
+
+#include <boost/bind.hpp>
+
+namespace graphene { namespace net_stats {
+
+namespace detail {
+
+class net_stats_impl {
+public:
+    net_stats_impl(net_stats_plugin& _plugin)
+        : _self(_plugin) 
+        {
+            // TODO ipaddr and port not static?
+            exposer  = std::make_shared<prometheus::Exposer>( "127.0.0.1:8080" );
```

Comment:
Can we have 127.0.0.1:8080 be a command line parameter, please?

Reply:
Yes, will do that.

Reply:
Added